src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.cs

```diff
@@ -62,6 +66,79 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .WithTrackingName(GenSpecTrackingName);
 
             context.RegisterSourceOutput(genSpec, ReportDiagnosticsAndEmitSource);
+
+            if (!s_hasInitializedInterceptorVersion)
+            {
+                InterceptorVersion = DetermineInterceptableVersion();
+                s_hasInitializedInterceptorVersion = true;
+            }
+        }
+
+        internal static int s_interceptorVersion;
+        internal static int InterceptorVersion
+        {
+            get
+            {
+                Debug.Assert(s_hasInitializedInterceptorVersion);
+                return s_interceptorVersion;
+            }
+            set
+            {
+                s_interceptorVersion = value;
+            }
+        }
+
+        // Used with v1 interceptor lightup approach:
+        private static bool s_hasInitializedInterceptorVersion;
+        internal static Func<SemanticModel, InvocationExpressionSyntax, CancellationToken, object>? GetInterceptableLocationFunc { get; private set; }
+        internal static MethodInfo? InterceptableLocationVersionGetDisplayLocation { get; private set; }
+        internal static MethodInfo? InterceptableLocationDataGetter { get; private set; }
+        internal static MethodInfo? InterceptableLocationVersionGetter { get; private set; }
+
+        internal static int DetermineInterceptableVersion()
+        {
+            MethodInfo? getInterceptableLocationMethod = null;
+            int? interceptableVersion = null;
+
+#if UPDATE_BASELINES
+            const string envVarName = "InterceptableAttributeVersion";
```

Comment:
>InterceptableAttributeVersion [](http://example.com/codeflow?start=39&length=29)

Is this specific only to the config binder source gen? or we expect usage in other places?

Reply:
Just here. I removed the const and just inlined the string.